packages/dbgpt-serve/src/dbgpt_serve/agent/agents/db_gpts_memory.py

```
@@ -79,7 +80,16 @@ def __init__(self):
 
     def append(self, message: GptsMessage):
-        self.gpts_message.append(message.to_dict())
+        data = message.to_dict()
+        # Link this message to its observability trace (for the Observability
+        # dashboard).
+        try:
+            span_id = root_tracer.get_current_span_id()
+            if span_id:
+                data["trace_id"] = span_id
+        except Exception:
+            pass
```

Ruff (0.16.2)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)